code/game/objects/structures/floodlight.dm

```diff
@@ -0,0 +1,322 @@
+			return
+
+		playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
+		if(do_mob(user,	src, 20))
+			if(anchored == 1)
+				if(!connect_to_network())
+					user << "<span class='notice'>You undo the ground bolts, freeing the [src].</span>"
+					anchored = 0
+					return
+				user << "<span class='notice'>You undo the ground bolts, freeing the [src]. It is no longer connected to the powernet.</span>"
+				anchored = 0
+				wiredtoground = 0
+				disconnect_from_network()
+				return
+			else
+				if(connect_to_network())
+					user << "<span class='notice'>You tighiten the ground bolts, bolting down the [src]. It has been connected to the powernet.</span>"
+					anchored = 1
+					wiredtoground = 1
+					return
+				user << "<span class='notice'>You tighiten the ground bolts, bolting down the [src].</span>"
+				anchored = 1
+			return
+		return
+
+	if(istype(I, /obj/item/weapon/screwdriver))
+		if(locked == 1)
+			user << "<span class='danger'>The controls are locked!</span>" //to prevent easy kills
+			return
+		if(cover == 1)
+			playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
+			user << "<span class='notice'>You open the maintenance hatch.</span>"
+			icon_state = "floodlight[status]_hatchopen[(powerpack ? 1 : 0)]"
+			cover = 0
+		else
+			playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
+			user << "<span class='notice'>You close the maintenance hatch.</span>"
+			icon_state = "floodlight[status]_hatchclose"
+			cover = 1
+		return
+
+	if(istype(I, /obj/item/weapon/card/emag))
+		message_admins("[user.ckey]([user]) emagged the floodlight at [src.loc]!")
+		playsound(loc, 'sound/effects/sparks1.ogg', 50, 1)
+		user << "<span class='danger'>You rig the floodlight to blow next time it is used!</span>"
+		locked = 0
+		rigged = 1
+
+	if(istype(I, /obj/item/weapon/crowbar))
+		if(cover == 0)
+			if(powerpack)
+				user << "<span class='danger'>Remove the cell first before deconstructing!</span>"
+			else
+				playsound(loc, 'sound/items/Crowbar.ogg', 50, 1)
+				user << "<span class='notice'>You begin deconstructing the [src]!</span>"
+				if(do_mob(user, src, 40))
+					var/obj/item/stack/sheet/metal/S = new
+					S.amount = 15
+					S.loc = user.loc
+					qdel(src)
+					return
+		return
+
+	if(I.force > 3)
+		user.visible_message("<span class='danger'>[user] hits the light with the [I]!</span>")
+		health = health - I.force
+		playsound(loc, 'sound/weapons/smash.ogg', 50, 1)
+		update_health()
+
+
+/obj/machinery/power/floodlight/attack_hand(mob/user)
+	if(broken)
+		user << "<span class='danger'>Its broken!</span>"
+		return
+	if(locked)
+		user << "<span class='danger'>The controls are locked!</span>"
+		return
+	if(wiredtoground == 1 && surplus() < 40 && !powerpack)
+		user << "<span class='danger'>There is no power in the connected powernet and no internal power source!</span>"
+		turnoff()
+		return
+	if(cover == 0)
+		if(powerpack == null)
+			return
+		powerpack.loc = get_turf(user)
```

Comment:
Can you maybe place the cell into the user's hand? Would be a nice quality of life thing